packages/e2e-tests/puppeteer/__tests__/web3torrent.test.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable jest/expect-expect */
+import {Page, Browser} from 'puppeteer';
+import {configureEnvVariables, getEnvBool} from '@statechannels/devtools';
+
+import {setUpBrowser, loadWeb3App, waitAndOpenChannel, waitForClosingChannel} from '../helpers';
+
+import {uploadFile, startDownload, cancelDownload} from '../scripts/web3torrent';
+
+jest.setTimeout(200_000);
+
+configureEnvVariables();
+const HEADLESS = getEnvBool('HEADLESS');
+
+let browserA: Browser;
+let browserB: Browser;
+let web3tTabA: Page;
+let web3tTabB: Page;
+
+describe('Supports torrenting among peers with channels', () => {
+  beforeAll(async () => {
+    // 100ms sloMo avoids some undiagnosed race conditions
+    browserA = await setUpBrowser(HEADLESS, 100);
+    browserB = await setUpBrowser(HEADLESS, 100);
+
+    web3tTabA = (await browserA.pages())[0];
+    web3tTabB = (await browserB.pages())[0];
+
+    await loadWeb3App(web3tTabA, 0, 'file/new', true);
+    await loadWeb3App(web3tTabB, 0, '', true);
```

Comment:
To speed things up a bit, you don't need to load web3torrent in tab B, since you will navigate away from the homepage later anyway.

Reply:
Actually we need that step to setup the browser for web3. So this step is needed.

Reply:
That step could be broken into separate functions so we can avoid the unnecessary page load.  I suppose it takes a little more refactoring to achieve, so we could tackle it in a future PR. Bottom line, we call `web3tTabB.goto` twice and we need only call it once. (It takes about 5s on my machine, so maybe it is not a big deal).

Reply:
Done. I took out `page.goto(...)` from the helper function.